### PR TITLE
Fix tab change logic

### PR DESCRIPTION
**Pull request by zombieJ**
* Fix tab shift when new tabPane has element with `autoFocus`
* Fix tab type `TAB` to make the focus out of TabPane

ref: https://github.com/ant-design/ant-design/issues/11748
ref: https://github.com/ant-design/ant-design/issues/11261

### CODE REVIEW

File: src/Tabs.js [hunk 1]
```diff
@@ -74,6 +74,12 @@ export default class Tabs extends React.Component {
     }
   }
 
+  onScroll = ({ target, currentTarget }) => {
+    if (target === currentTarget && target.scrollLeft > 0) {
+      target.scrollLeft = 0;
+    }
```
Comment (nezed) [inline src/Tabs.js]: Looks like this solution includes experience of https://github.com/react-component/tabs/issues/133
Thank you)

File: src/TabPane.js [hunk 2]
```diff
@@ -18,6 +49,30 @@ export default class TabPane extends React.Component {
...
+      );
+      sentinelEnd = (
+        <div
+          tabIndex={0}
+          ref={this.setSentinelEnd}
+          style={sentinelStyle}
+          onKeyDown={this.onBottomSentienelKeyDown}
+        />
+      );
+    }
```
Comment (afc163) [inline src/TabPane.js]: 这两个 div 会不会影响现有的 tab 顺序？

File: src/TabPane.js [hunk 3]
```diff
@@ -18,6 +49,30 @@ export default class TabPane extends React.Component {
...
+    let sentinelStart;
+    let sentinelEnd;
+    if (shouldRender) {
+      const sentinelStyle = { width: 0, height: 0, overflow: 'hidden' };
+      sentinelStart = (
+        <div
+          tabIndex={0}
+          ref={this.setSentinelStart}
+          style={sentinelStyle}
+          onKeyDown={this.onTopSentienelKeyDown}
```
Comment (afc163) [inline src/TabPane.js]:
```diff
- Sentienel
- Sentinel
```

File: src/TabPane.js [hunk 4]
```diff
@@ -1,9 +1,40 @@
...
+  onBottomSentienelKeyDown = ({ target, which, shiftKey }) => {
+    if (
+      this.sentinelStart &&
+      which === KeyCode.TAB &&
+      document.activeElement === target &&
+      !shiftKey
+    ) {
+      this.sentinelStart.focus();
+    }
+  };
```
Comment (afc163) [inline src/TabPane.js]: 没太看懂，能解释一下这两个元素的作用么

File: src/TabPane.js (hunk 2 above)
Comment (zombieJ) [reply src/TabPane.js]: 不会，这两个是防止用户在 TabPane 里的最后一个元素再按 Tab 切到视图外的 TabPane 里导致错误。如果 sentinel 发现用户在这个元素上又 tab 了一次，到另一个 sentinel 上。这样 tab 和 shift tab 就都保持在 TabPane 里了：
### tab:
kapture 2018-11-02 at 11 26 19

### shift tab:
kapture 2018-11-02 at 11 27 06

Comment (afc163) [reply src/TabPane.js]: 这样不是永远困在这个 pane 里面了？

如果想要继续 tab 到 Tabs 外怎么办

Comment (zombieJ) [reply src/TabPane.js]: 也有道理，在最外面再加 2 个 sentinel。这样 tab 就可以出去了。

Comment (afc163) [reply src/TabPane.js]: 我感觉这个思路会把事情搞复杂。

**Comment by afc163** [comment]
https://github.com/ant-design/ant-design/issues/11261 这个问题已经不存在了。

只需要修复 https://github.com/ant-design/ant-design/issues/11748 即可。

**Comment by nezed** [comment]
Тоже напишут тут на не понятном для вас языке
А вообще я недоволен и уверен что решение получится  ещё более багованым

**Comment by loonywizard** [comment]
guys would you be so kind to write all messages in english so everyone can understand what are you talking about?

**Comment by zombieJ** [comment]
Hi @nezed  @loonywizard ,
Thanks for your attention on this issue. Current issue is focus on the tab interactive in panel interactive. To resolve user tap tab in tab pane cause focus to the wrong focusable element which in the hidden tab pane. So I add the sentinel element to detect the focus state. If the sentinel get focus and user tap tab, this will pass focus to another sentinel and pass tab to out of the tab.

**Comment by erixtekila** [comment]
+1 @zombieJ 
I've got the same issue.

**Pull request merged**